gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -215,39 +215,39 @@ protected GoogleCloudStorageItemInfo getInitialMetadata() throws IOException {
     return null;
   }
 
-  /** Returns {@link GoogleCloudStorageItemInfo} used to initialize metadata in constructor. */
-  private GoogleCloudStorageItemInfo fetchInitialMetadata() throws IOException {
-    StorageObject object;
+  /** Method used to cache footer and initializes metadata in constructor. */
+  private void prefetchFooterAndInitMetadata() throws IOException {
+    checkState(!metadataInitialized, "metadata should not be initialized yet for '%s'", resourceId);
+
+    // TODO: use metadata request if `readOptions.getMinRangeRequestSize()` is 0.
+    Storage.Objects.Get getObject =
+        createDataRequest(/* rangeHeader= */ "bytes=-" + readOptions.getMinRangeRequestSize());
```

Comment:
Previously, the metadata request used `setFields` to select only the required metadata. Is it possible to preserve that behavior on this data request?

Reply:
Fields filtering applicable only for metadata requests that return object metadata in JSON format in HTTP response body. It's not applicable to data request used for footer pre-fetching as it returns raw bytes in HTTP body and we take necessary object metadata from HTTP headers instead that already very limited and can not be filtered.